### PR TITLE
Fix label fields validation, fix max length validation

### LINKED ISSUE #1342: Label fields must not require input
This seems to be a regression, so it definitely needs tests. Also the example form needs one of each type of field.

### PR DESCRIPTION
**Pull request by luontola**
Closes #1342

### CODE REVIEW

File: tests.edn [hunk 1]
```diff
@@ -9,7 +9,11 @@
                     {:id :integration
                      :ns-patterns ["rems"]
                      :test-paths ["src" "test"]
-                     :focus-meta [:integration]}]
+                     :focus-meta [:integration]}
+                    {:id :focused
```
Comment (foxlynx) [inline tests.edn]: Should this be in some other pull request? Can't see anywhere focused appended elsewhere in this PR?

File: test/clj/rems/migrations/test_convert_to_dynamic_applications.clj [hunk 2]
```diff
@@ -3,13 +3,12 @@
             [clojure.pprint :refer [pprint]]
```
Comment (foxlynx) [inline test/clj/rems/migrations/test_convert_to_dynamic_applications.clj]: Also don't get how this file is relevant to field changes?

File: tests.edn (hunk 1 above)
Comment (luontola) [reply tests.edn]: The :focused tag is not meant to be committed. It's for easily choosing which subset of tests to run.

File: test/clj/rems/migrations/test_convert_to_dynamic_applications.clj (hunk 2 above)
Comment (luontola) [reply test/clj/rems/migrations/test_convert_to_dynamic_applications.clj]: The test expectations must be changed when the label field was added to the example form.

File: tests.edn (hunk 1 above)
Comment (foxlynx) [reply tests.edn]: Still don't get why it is added in this pull request.

**Pull request merged**